ci/tests/vmachine/vmachine_test.py

```diff
@@ -184,3 +223,86 @@ def vms_with_fio_test():
+
+
+def set_vm_as_template_test():
+    """
+    {0}
+    """.format(general.get_function_name())
+
+    general.check_prereqs(testcase_number=2,
+                          tests_to_run=testsToRun)
+
+    api = Connection.get_connection()
+    vpool_list = api.get_component_by_name('vpools', VPOOL_NAME)
+    assert len(vpool_list), "No vpool found where one was expected"
+    vpool = vpool_list[0]
+
+    create_raw_disk(vpool['name'], TEMPLATE_DISK_NAME)
+    create_machine_from_existing_raw_disk(TEMPLATE_MACHINE_NAME, vpool['name'], TEMPLATE_DISK_NAME)
+
+    counter = TIMEOUT / TIMER_STEP
+    while counter > 0:
+        vmachine_list = api.get_component_by_name('vmachines', TEMPLATE_MACHINE_NAME)
+        if vmachine_list:
+            counter = 0
+        else:
+            counter -= 1
+            time.sleep(TIMER_STEP)
+
+    shut_down_vmachine_by_name(TEMPLATE_MACHINE_NAME)
+
+    counter = TIMEOUT / TIMER_STEP
+    while counter > 0:
+        vmachine_list = api.get_component_by_name('vmachines', TEMPLATE_MACHINE_NAME)
+        if vmachine_list[0]['hypervisor_status'] == 'TURNEDOFF':
+            counter = 0
+        else:
+            counter -= 1
+            time.sleep(TIMER_STEP)
+    vmachine_list = api.get_component_by_name('vmachines', TEMPLATE_MACHINE_NAME)
+    assert vmachine_list[0]['hypervisor_status'] == 'TURNEDOFF', \
+        "VMachine {0} expected to be in status TURNEDOFF, instead found in status {1} after {2} seconds".format(TEMPLATE_MACHINE_NAME,
+                                                                                                                vmachine_list[0]['hypervisor_status'],
+                                                                                                                TIMEOUT)
+    VMachineController.set_as_template(vmachine_list[0]['guid'])
+
+    vmachine_list = api.get_component_by_name('vmachines', TEMPLATE_MACHINE_NAME)
+    assert vmachine_list[0]['is_vtemplate'], 'Vmachine {0} is not a template'.format(TEMPLATE_MACHINE_NAME)
+
+    vdisk_list = VDiskList.get_vdisk_by_name(TEMPLATE_DISK_NAME)
+    if vdisk_list and len(vdisk_list):
+        vdisk_list[0].invalidate_dynamics(['is_vtemplate'])
+
+    vdisks = api.get_component_by_name('vdisks', TEMPLATE_DISK_NAME)
+    assert vdisks[0]['is_vtemplate'], "VDisk {0} expected to be changed to template after {1} seconds".format(TEMPLATE_DISK_NAME, TIMEOUT)
+
+
+def create_vm_from_template_test():
+    """
+    {0}
+    """.format(general.get_function_name())
+
+    general.check_prereqs(testcase_number=3,
+                          tests_to_run=testsToRun)
+
+    api = Connection.get_connection()
+    template_machine_list = api.get_component_by_name('vmachines', TEMPLATE_MACHINE_NAME)
+    if template_machine_list is None or len(template_machine_list) == 0:
+        raise SkipTest('Template vmachine not found on environment')
```

Comment:
raise should not be done here imo
If we allow to run specific tests, the tests should independent of eachother
meaning, if you want to run this specific test, the test should check for presence of a template, if no template present --> create 1